docs/cli/tkgctl/bom.md

```diff
@@ -0,0 +1,124 @@
+## Introduction
+
+BoM stands for Bill of Materials and there are 2 types of BoM files:
+
+* TKG BoM file   (1 file per release)
+* TKR BoM files (multiple files per release)
+
+### TKG BoM file
+
+* contains information about TKG related component which mainly gets associated with management cluster creation
+  * This file includes the TKG version under release.version key and default TKR version under default.k8sVersion key
+  * This includes cluster-api provider components, pinniped components, image repo information, and many more.
+  * [A sample TKG BoM file](example-boms/tkg-bom.yaml)
+
+### TKR BoM file
+
+* this file contains information related to
+  * specific k8s version
+  * node images to use with (vsphere, aws, azure) for the matching k8s version
+  * add-ons specific components
+  * [A sample TKR BoM file](example-boms/tkr-bom.yaml)
+
+## Bundling BoM files into TKG library
+
+* Tanzu CLI is decoupled from TKG BoM file (and TKR BoM file) and won't be bundled in CLI library.
+* Tanzu CLI is compiled with TKG Compatibility Image path( build time constant ). Tanzu CLI at runtime downloads the compatibility file and determines the TKG BoM image path from the compatibility file (version matching the management-cluster plugin version currently installed) and downloads the BoM files as well to the user's machine. [reference](../../../tkg/tkgconfigupdater/ensure.go)
+
+### Why do we need TKG Compatibility Image?
+
+If the BoM files are bundled with CLI, and when there is a CVE fix in any images/components that are part of TKG/TKR BoM file for an existing release, this would warrant a new CLI release.
+By using a TKG Compatibility Image, a new CLI release can be avoided. The release team can publish a new compatibility image(new Tag) with updated TKG BoM paths for a given management-cluster plugin version, and CLI would download the new Compatibility Image(with the latest Image Tag) and there by can download and use the updated TKG/TKR BoM files while creating a new management cluster.
+
+Sample compatibility file is as shown below:
+
+```yaml
+version: v2
+managementClusterPluginVersions:
+- version: v1.5.1
+  supportedTKGBomVersions:
+  - imagePath: tkg-bom
+    tag: v1.5.1
+- version: v1.5.0
+  supportedTKGBomVersions:
+  - imagePath: tkg-bom
+     tag: v1.5.0
+```
+
+Sample compatibility file after the updating the BoM version(new BoM which has images/components with CVE fixes for v1.5.0) is shown below. The new compatibility image would be pushed to image repository with updated tag( eg: **tkg/tkg-compatibility:v3**)
+
+```yaml
+version: v3
+managementClusterPluginVersions:
+  - version: v1.5.1
+    supportedTKGBomVersions:
+      - imagePath: tkg-bom
+        tag: v1.5.1
+  - version: v1.5.0
+    supportedTKGBomVersions:
+      - imagePath: tkg-bom
+          tag: v1.5.0-patch
+```
+
+Notes:
+
+* Once we create a management cluster it is running the TKR controller which reconciles all the supported TKR and updates TKR compatibility.
```

Comment:
I'm not clear on what this means: "reconciles all the supported TKR and updates TKR compatibility".  Are we talking about "TKR BoM files and TKR Compatibility Image"?

Reply:
TKR controller reconciles the TKR BoM images and TKR compatibility Image from Image registry and later creates configmaps and TKR resources in management cluster using these images.  

I will update this to give more details.